pkg/tests/tasks/traffic/ingress/gatewayapi_test.go

```diff
@@ -0,0 +1,205 @@
+package ingress
+
+import (
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+func TestGatewayApi(t *testing.T) {
+	NewTest(t).Id("T41").Groups(Full, InterOp).Run(func(t TestHelper) {
+		if env.GetSMCPVersion().LessThan(version.SMCP_2_3) {
+			t.Skip("TestGatewayApi was added in v2.3")
+		}
+		ns := "foo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
+			oc.RecreateNamespace(t, ns)
+		})
+
+		var SMCP_NAME = "basic"
+
+		ossm.DeployControlPlane(t)
+
+		t.LogStep("Install Gateway API CRD's")
+		shell.Executef(t, "kubectl get crd gateways.gateway.networking.k8s.io &> /dev/null && echo 'Gateway API CRDs already installed' || kubectl apply -k github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v0.5.1")
+
+		oc.CreateNamespace(t, ns)
+
+		t.NewSubTest("Deploy the Kubernetes Gateway API").Run(func(t test.TestHelper) {
+
+			t.Cleanup(func() {
+				oc.RecreateNamespace(t, meshNamespace)
+				oc.RecreateNamespace(t, ns)
+			})
+
+			t.LogStep("Install httpbin")
+			app.InstallAndWaitReady(t, app.Httpbin(ns))
+
+			t.LogStep("Deploy the Gateway SMCP")
+
+			if env.GetSMCPVersion().LessThan(version.SMCP_2_4) {
+				oc.Patch(t, meshNamespace, "smcp", SMCP_NAME, "merge", `
+        spec:
+          runtime:
+            components:
+              pilot:
+                container:
+                  env:
+                    PILOT_ENABLE_GATEWAY_API: “true”
+                    PILOT_ENABLE_GATEWAY_API_STATUS: “true”
+                    PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER: “true”`)
+
+				t.Cleanup(func() {
```

Comment:
I think can be better to define only one clean func to this and put in a func called deleteGatewaySMCP or something like that and make the logic inside that func to define which version will be deleted